br/pkg/lightning/backend/local/local.go

```diff
@@ -924,30 +946,32 @@ func splitRangeBySizeProps(fullRange Range, sizeProps *sizeProperties, sizeLimit
 	curSize := uint64(0)
 	curKeys := uint64(0)
 	curKey := fullRange.start
+
 	sizeProps.iter(func(p *rangeProperty) bool {
-		if bytes.Equal(p.Key, engineMetaKey) {
+		if bytes.Compare(p.Key, fullRange.start) <= 0 {
 			return true
 		}
+		if bytes.Compare(p.Key, fullRange.end) > 0 {
+			return false
+		}
 		curSize += p.Size
 		curKeys += p.Keys
-		if int64(curSize) >= sizeLimit || int64(curKeys) >= keysLimit {
-			// in case the sizeLimit or keysLimit is too small
-			endKey := p.Key
-			if bytes.Equal(curKey, endKey) {
-				endKey = nextKey(endKey)
-			}
-			ranges = append(ranges, Range{start: curKey, end: endKey})
-			curKey = endKey
+		if bytes.Compare(p.Key, curKey) > 0 && (int64(curSize) >= sizeLimit || int64(curKeys) >= keysLimit) {
```

Comment:
does `bytes.Compare(p.Key, curKey) > 0` redundant here since `bytes.Compare(p.Key, fullRange.start)` is already checked and the properties are sorted by key